src/main/java/com/google/codeu/servlets/MessageServlet.java

```diff
@@ -96,51 +84,21 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 
     String user = userService.getCurrentUser().getEmail();
     String text = Jsoup.clean(request.getParameter("text"), Whitelist.relaxed());
+    String textWithMedia = getMediaEmbeddedText(text);
 
-    String regexImage = "(https?://\\S+\\.(png|jpg|gif))";
-    String replacementImage = "<img src=\"$1\" />";
-    String regexVideo = "(https?://www.youtube.com/\\S+)";
-    String replacementVideo = "<iframe width=\"420\" height=\"345\" src=\"$1\">" + "</iframe>";
-
-    String textForImageAndVideo = text;
-
-    // Validation of URL
-    Pattern patternImg = Pattern.compile(regexImage);
-    Pattern patternVid = Pattern.compile(regexVideo);
-    Matcher matcherImg = patternImg.matcher(text);
-    Matcher matcherVid = patternVid.matcher(text);
-
-    // Checks if the URL is valid and if it´s then it changes to insert the image
-    if (matcherImg.find() && urlValidator(matcherImg.group())) {
-      textForImageAndVideo = text.replaceAll(regexImage, replacementImage);
-    }
-
-    // Checks if the URL is valid and if it´s then it changes to insert the video
-    if (matcherVid.find() && urlValidator(matcherVid.group())) {
-      // Change the format of the normal Youtube URL to an embed one
-      textForImageAndVideo = textForImageAndVideo.replace("watch?v=", "embed/");
-      textForImageAndVideo = textForImageAndVideo.replaceAll(regexVideo, replacementVideo);
-    }
-
-    // get value of query parameter
-    String recipient = request.getParameter("recipient");
-
-    String regex = "(https?://\\S+\\.(png|jpg))";
-    String replacement = "<img src=\"$1\" />";
-    String textWithImagesReplaced = text.replaceAll(regex, replacement);
-
-    float sentimentScore = this.getSentimentScore(text);
+    String country = request.getParameter("country");
 
-    // check if user is valid
-    if (datastore.getUser(recipient) == null) {
-      response.sendRedirect("/users/" + user);
+    // Redirect to home on invalid country
```

Comment:
So, if the user somehow tried to post a message to the page of a country that doesn't exist. As the code stands right now, I don't think this is possible (Because non-existent countries don't have pages, and attempts to access the page through localhost:8080/country/[doesn't exist] are redirected to the invalid-country error page, unless the user would somehow alter the page scripts themselves. It may be an extra check, as I'm not sure that the user could ever change that form, but it's the same thing as we had of checking if users tried to post something to a non-existant user page

Long story short, yes, we do generate the country pages; but they are stored similarly to users in datastore. Therefore, we can easily check if a user is trying to send a message to a country that they doesn't exist by checking if we have data on this country